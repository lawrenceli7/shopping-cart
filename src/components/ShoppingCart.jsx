import styled from "styled-components";

const StyledShoppingCart = styled.div`
  border: 1px solid #ccc;
  padding: 16px;
  margin: 16px;
  text-align: center;

  ul {
    list-style-type: none;
    padding: 0;
  }

  button {
    font-size: 15px;
    padding: 10px;
    margin: 1px;
    border: none;
    background-color: #f0f0f0;
    cursor: pointer;
  }
`;

const ShoppingCart = ({ cart }) => {
  const getTotal = () => {
    return cart
      .reduce((sum, item) => sum + item.price * item.quantity, 0)
      .toFixed(2);
  };

  return (
    <StyledShoppingCart>
      <h2>Shopping Cart</h2>
      {cart.length === 0 ? (
        <p>Your cart is empty</p>
      ) : (
        <ul>
          {cart.map((item, index) => (
            <li key={index}>
              {item.title} - {item.quantity} x ${item.price}
            </li>
          ))}
        </ul>
      )}
      <h3>Total: ${getTotal()}</h3>
      <button>Checkout</button>
    </StyledShoppingCart>
  );
};

export default ShoppingCart;
