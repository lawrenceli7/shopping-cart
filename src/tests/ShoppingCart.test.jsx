import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import ShoppingCart from "../components/ShoppingCart";

describe("ShoppingCart Component", () => {
  it("should render the shopping cart title", () => {
    const { asFragment } = render(<ShoppingCart cart={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should display a message when the cart is empty", () => {
    const { asFragment } = render(<ShoppingCart cart={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render items in the cart", () => {
    const cartItems = [
      { title: "Item 1", price: 10, quantity: 2 },
      { title: "Item 2", price: 15, quantity: 1 },
    ];
    const { asFragment } = render(<ShoppingCart cart={cartItems} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should calculate and display the total price", () => {
    const cartItems = [
      { title: "Item 1", price: 10, quantity: 2 },
      { title: "Item 2", price: 15, quantity: 1 },
    ];
    const { asFragment } = render(<ShoppingCart cart={cartItems} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render the checkout button", () => {
    const { asFragment } = render(<ShoppingCart cart={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
