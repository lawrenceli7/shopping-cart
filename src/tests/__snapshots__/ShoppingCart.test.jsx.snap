// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ShoppingCart Component > should calculate and display the total price 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered sc-aYaIB iKlttL css-dev-only-do-not-override-3rel02"
  >
    <div
      class="ant-card-body"
    >
      <h2>
        Shopping Cart
      </h2>
      <ul>
        <li>
          Item 1 - 2 x $10
        </li>
        <li>
          Item 2 - 1 x $15
        </li>
      </ul>
      <h3>
        Total: $35.00
      </h3>
      <button
        class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default"
        type="button"
      >
        <span>
          Checkout
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ShoppingCart Component > should display a message when the cart is empty 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered sc-aYaIB iKlttL css-dev-only-do-not-override-3rel02"
  >
    <div
      class="ant-card-body"
    >
      <h2>
        Shopping Cart
      </h2>
      <p>
        Your cart is empty
      </p>
      <h3>
        Total: $0.00
      </h3>
      <button
        class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default"
        type="button"
      >
        <span>
          Checkout
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ShoppingCart Component > should render items in the cart 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered sc-aYaIB iKlttL css-dev-only-do-not-override-3rel02"
  >
    <div
      class="ant-card-body"
    >
      <h2>
        Shopping Cart
      </h2>
      <ul>
        <li>
          Item 1 - 2 x $10
        </li>
        <li>
          Item 2 - 1 x $15
        </li>
      </ul>
      <h3>
        Total: $35.00
      </h3>
      <button
        class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default"
        type="button"
      >
        <span>
          Checkout
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ShoppingCart Component > should render the checkout button 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered sc-aYaIB iKlttL css-dev-only-do-not-override-3rel02"
  >
    <div
      class="ant-card-body"
    >
      <h2>
        Shopping Cart
      </h2>
      <p>
        Your cart is empty
      </p>
      <h3>
        Total: $0.00
      </h3>
      <button
        class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default"
        type="button"
      >
        <span>
          Checkout
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ShoppingCart Component > should render the shopping cart title 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered sc-aYaIB iKlttL css-dev-only-do-not-override-3rel02"
  >
    <div
      class="ant-card-body"
    >
      <h2>
        Shopping Cart
      </h2>
      <p>
        Your cart is empty
      </p>
      <h3>
        Total: $0.00
      </h3>
      <button
        class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default"
        type="button"
      >
        <span>
          Checkout
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
